lib/crypto/packaged.py

```diff
@@ -51,6 +52,17 @@ def get_endpoint(server):
     return u'{server}/1.0/sign_addon'.format(server=server)
 
 
+def get_id(addon):
+    """Return the addon GUID if <= 64 chars, or its sha256 hash otherwise.
+
+    We don't want GUIDs longer than 64 chars: bug 1203365.
+    """
+    guid = addon.guid
+    if len(guid) <= 64:
+        return guid
+    return hashlib.sha256(guid).hexdigest()
```

Comment:
We could always `'sha256:%s' % hash[len('sha256:'):]` (surely there's a less ugly way to do that) if necessary :) But fortunately I don't think it's necessary. Might be nice for someone looking at the cert, granted, but I'd rather not lose the 56 bits of entropy if we can help it.